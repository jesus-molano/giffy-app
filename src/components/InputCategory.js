import { useState } from 'react';
import PropTypes from 'prop-types';

const InputCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');
	const handleInputChange = (e) => setInputValue(e.target.value);
	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 0) {
			setCategories((category) => [ inputValue,...category]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<input
				type='text'
				value={inputValue}
				onChange={handleInputChange}
				placeholder="Search Gifs Here..."
			></input>
		</form>
	);
};

InputCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};

export default InputCategory;
