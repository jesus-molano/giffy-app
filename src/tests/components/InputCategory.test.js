import { shallow } from 'enzyme';
import InputCategory from '../../components/InputCategory';

describe('InputCategory component tests', () => {

  const setCategories =  jest.fn();
  let wrapper = shallow(<InputCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<InputCategory setCategories={setCategories} />)

  })
  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  test('should change the text box', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo'
    input.simulate('change', {target: {value}})
  })
  test('should NOT post the info with submit', () =>{
    wrapper.find('form').simulate('submit', { preventDefault() { } });
    expect(setCategories).not.toHaveBeenCalled();
  })
  test('should call the setCategories and clean the input', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    // Simular rellenar el input
    input.simulate('change', { target: { value } });
    // Simular hacer submit
    wrapper.find('form').simulate('submit', { preventDefault() { } });
    // Espera que la funcion setCategories sea llamada
    expect(setCategories).toHaveBeenCalled();
    // El input ahora estara vacio
    expect(input.prop('value')).toBe('');
  })
  
})
