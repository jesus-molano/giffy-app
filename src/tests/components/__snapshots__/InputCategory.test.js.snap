// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCategory component tests should show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search Gifs Here..."
    type="text"
    value=""
  />
</form>
`;
